contracts/governance/Staking/modules/StakingVestingModule.sol

```diff
@@ -0,0 +1,321 @@
+pragma solidity ^0.5.17;
+pragma experimental ABIEncoderV2;
+
+import "./shared/StakingShared.sol";
+import "../../../proxy/modules/interfaces/IFunctionsList.sol";
+
+/**
+ * @title Staking Vesting Module contract
+ * @notice Implements interaction with Vesting functionality: vesting registry, vesting staking
+ * */
+contract StakingVestingModule is IFunctionsList, StakingShared {
+    event ContractCodeHashAdded(bytes32 hash);
+    event ContractCodeHashRemoved(bytes32 hash);
+    event VestingStakeSet(uint256 lockedTS, uint96 value);
+
+    /**
+     * @notice sets vesting registry
+     * @param _vestingRegistryProxy the address of vesting registry proxy contract
+     * @dev _vestingRegistryProxy can be set to 0 as this function can be reused by
+     * various other functionalities without the necessity of linking it with Vesting Registry
+     */
+    function setVestingRegistry(address _vestingRegistryProxy) external onlyOwner whenNotFrozen {
+        vestingRegistryLogic = IVestingRegistry(_vestingRegistryProxy);
+    }
+
+    /**
+     * @notice Sets the users' vesting stakes for a giving lock dates and writes checkpoints.
+     * @param lockedDates The arrays of lock dates.
+     * @param values The array of values to add to the staked balance.
+     */
+    function setVestingStakes(uint256[] calldata lockedDates, uint96[] calldata values)
+        external
+        onlyAuthorized
+        whenNotFrozen
+    {
+        require(lockedDates.length == values.length, "WS05"); // arrays mismatch
+
+        uint256 length = lockedDates.length;
+        for (uint256 i = 0; i < length; i++) {
+            _setVestingStake(lockedDates[i], values[i]);
+        }
+    }
+
+    /**
+     * @notice Sets the users' vesting stake for a giving lock date and writes a checkpoint.
+     * @param lockedTS The lock date.
+     * @param value The value to be set.
+     */
+    function _setVestingStake(uint256 lockedTS, uint96 value) internal {
```

Comment:
I think these functions were only required when we decided to track vesting stakes separately from voluntary stakes. Back then, we needed to tell the contract which of the existing stakes correspond to a vesting contract. Now, we should have no need for it anymore and should consider removing them. thoughts?

Reply:
@tjcloa

Reply:
keeping it for now
added TODO to deprecate